pkg/extender/extender.go

```diff
@@ -715,6 +723,8 @@ sendResponse:
 		score, ok := priorityMap[node.Name]
 		if !ok || score == 0 {
 			score = int(defaultScore)
+		} else if score == -1 {
+			score = int(invalidNodeScore)
```

Comment:
nit..We assign a score of 0 to offline nodes and later set it to defaultScore (5).
We assign -1 to invalid nodes and later set it to invalidNodeScore (0).

Instead shouldn't we just update a score of 5 to offline nodes and 0 to invalid nodes and just use those values from the priorityMap here? I think it would be just a bit cleaner. Wdyt?

Reply:
In the current implementation priorityMap is initialized with all values set to 0.
What you are saying makes sense however if we initialise the map with score of 5, we will have to make changes in unit tests and integration tests wherever required (since node which earlier had a score of 2*nodePriorityScore will have a score of 2*nodePriorityScore+defaultScore now and so on)

Since we want to get this change in on a tight deadline I feel it makes more sense to address this later

Reply:
Makes sense.
But this won't necessarily make the code cleaner.
1. The below check will still have to remain 
```
if !ok {
score = int(defaultScore)
}
```

2. In line # 595 , we will have to handle updating the 0 score (assigned to every node when priorityMap was initialized) to defaultScore here separately.
```
if disableHyperconvergence {
		goto sendResponse
	}
```

3. Also making that change will involve calculating and changing unit tests and integration tests as well since the final scores for nodes will change a bit.

For example, 
```
priorityMap[node.Name] += int(e.getNodeScore(node, volume, &rackInfo, &zoneInfo, &regionInfo, storageNode))
```
If we return defaultScore instead of 0 from e.getNodeScore(), then the final scores will vary based on the setup because we are adding the scores.

Reply:
Yeah. Let's not complicate things for now.